lib/tests/streamlit/Server_test.py

```diff
@@ -131,6 +134,58 @@ def test_healthz(self):
         self.assertEqual(503, response.code)
 
 
+class PortRotateAHundredTest(unittest.TestCase):
+    """Tests port rotates up to a hundred times then sys exits"""
+
+    def get_app(self):
+        app = mock.MagicMock()
+
+        def listen(port):
+            raise OSError(errno.EADDRINUSE, "test", "asd")
+
+        app.listen = listen
+        return app
+
+    def test_rotates_a_hundred_ports(self):
+        app = self.get_app()
+        with pytest.raises(SystemExit) as pytest_wrapped_e:
+            start_listening(app)
+            assert pytest_wrapped_e.type == SystemExit
+            assert pytest_wrapped_e.value.code == 42
+
+        assert app.listen.call_count == 100
```

Comment:
the problems I'm having are 2, first I'd want to check that this function had a hundred calls